src/cookbook/serverless-blog.md

```diff
@@ -0,0 +1,349 @@
+:::
+
+::: warning
+Using the component this way will mean that the lifecycle hooks of the component will not be called. Visit the Vue Router's docs to learn more about [Dynamic Route Matching](https://router.vuejs.org/en/essentials/dynamic-matching.html)
+:::
+
+To fix this we need to watch the `route.params.slug` value and call `getPost()` when it changes.
+
+Update `<script>` section in `views/BlogPost.vue`:
+
+```html{26,41}
+<template>
+<div
+    id="blog-post"
+    v-if="loaded"
+>
+    <h1>{{ post.data.title }}</h1>
+    <h4>{{ post.data.author.first_name }} {{ post.data.author.last_name }}</h4>
+    <div v-html="post.data.body"></div>
+
+    <router-link
+        v-if="post.meta.previous_post"
+        :to="{ name: 'BlogPost', params: { slug: post.meta.previous_post.slug } }"
+    >
+        {{ post.meta.previous_post.title }}
+    </router-link>
+    <router-link
+        :to="{ name: 'BlogPost', params: { slug: post.meta.next_post.slug } }"
+        v-if="post.meta.next_post"
+    >
+        {{ post.meta.next_post.title }}
+    </router-link>
+</div>
+</template>
+
+<script>
+import { defineComponent, onMounted, ref, unref, watch } from "vue"
+import { RouterLink, useRoute } from "vue-router"
+
+import { butter } from "@/buttercms.js"
+
+export default defineComponent({
+    name: "BlogPost",
+    components: {
+        RouterLink
+    },
+    setup () {
+        const post = ref([])
+        const loaded = ref( false )
+        const route = useRoute()
+
+        watch( () => route.params.slug, ( postSlug ) => getPost( postSlug ) )
+
+        onMounted( () => getPost( route.params.slug ) )
+
+        async function getPost ( postSlug ) {
+            loaded.value = false
+            post.value = (
+                await butter.post.retrieve( postSlug )
+            ).data
+            document.title = unref( post ).data.title
+            loaded.value = true
+        }
+
+        return {
+            loaded,
+            post,
+        }
+    },
+})
+</script>
+```
+
+Now your app has a working blog that can be updated easily in the [Blog posts ButterCMS dashboard](https://buttercms.com/blog_home/).
+
+## Categories, Tags, and Authors
+
+Use Butter's APIs for categories, tags, and authors to feature and filter content on your blog.
+
+See the ButterCMS API reference for more information about these objects:
+
+* [Categories](https://buttercms.com/docs/api/?javascript#categories)
+* [Tags](https://buttercms.com/docs/api/?javascript#tags)
+* [Authors](https://buttercms.com/docs/api/?javascript#authors)
+
+Here's an example of listing all categories and getting posts by category. Call these methods on the `onMounted` hook inside `setup`:
+
+```javascript
+onMounted( async () => {
+    const categories = await getCategories()
+    const postsByCategory = await getPostsByCategory( "example-category" )
+
+    console.log(categories, postsByCategory)
```

Comment:
Lets remove the `console.log` unless we are describing why thats in the code. if it should remain, then lets do:

`console.log({ categories, postsByCategory })`

which will output:

```
{
  categories: [ ... ],
  postsByCategory: [ ... ]
}
```

this is easier in the console to tell what data you are looking at.